apps/meteor/ee/app/livechat-enterprise/server/services/omnichannel.internalService.ts

```diff
@@ -23,13 +24,20 @@ export class OmnichannelEE extends ServiceClassInternal implements IOmnichannelE
 	}
 
 	async placeRoomOnHold(
-		room: Pick<IOmnichannelRoom, '_id' | 't' | 'open' | 'onHold'>,
+		room: Pick<IOmnichannelRoom, '_id' | 't' | 'open' | 'onHold' | 'u' | 'lastMessage'>,
 		comment: string,
 		onHoldBy: Pick<IUser, '_id' | 'username' | 'name'>,
 	) {
 		this.logger.debug(`Attempting to place room ${room._id} on hold by user ${onHoldBy?._id}`);
 
 		const { _id: roomId } = room;
+		const restrictedOnHold = settings.get('Livechat_allow_manual_on_hold_upon_agent_engagement_only');
+		const canRoomBePlacedOnHold = !room.onHold;
+		const canAgentPlaceOnHold = !room.lastMessage.token;
+		const canPlaceChatOnHold = canRoomBePlacedOnHold && (!restrictedOnHold || canAgentPlaceOnHold);
+		if (!canPlaceChatOnHold) {
+			throw new Error('error-contact-sent-last-message-so-cannot-place-on-hold');
+		}
```

Comment:
Why are you checking all of these conditions _before_ checking for room to be a valid Omnichannel room or room being not open?

These should happen after.